plugin/src/test/java/hudson/plugins/swarm/test/TestUtils.java

```diff
@@ -0,0 +1,92 @@
+package hudson.plugins.swarm.test;
+
+import static org.hamcrest.Matchers.greaterThan;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.io.ByteStreams;
+import hudson.Util;
+import hudson.model.Computer;
+import hudson.model.Node;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URL;
+import jenkins.model.Jenkins;
+import org.jvnet.hudson.test.JenkinsRule;
+
+/** Utilities for testing the Swarm Plugin */
+public class TestUtils {
+
+    /** Download the Swarm Client from the given Jenkins URL into the given temporary directory. */
+    private static File download(URL jenkinsUrl, File tmpDir) throws Exception {
+        assertTrue(tmpDir.isDirectory());
+
+        URL input =
+                jenkinsUrl.toURI().resolve(new URI(null, "swarm/swarm-client.jar", null)).toURL();
+        File output = new File(tmpDir, "swarm-client.jar");
+        try (InputStream inputStream = input.openStream();
+                FileOutputStream outputStream = new FileOutputStream(output)) {
+            ByteStreams.copy(inputStream, outputStream);
```

Comment:
I think there i a `FileUtils.copy` or similar that makes that simpler.

Reply:
Done.